### PR TITLE
Fix inherited private fields

### PR DESCRIPTION
**Pull request by EamonNerbonne**

### CODE REVIEW

**Comment by EamonNerbonne** [comment]
@charnoldus - this should fix the inherited private field issue...

File: ValueUtils/ReflectionHelper.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ValueUtils
+{
+    static class ReflectionHelper
+    {
+        public static IEnumerable<Type> WalkMeaningfulInheritanceChain(Type type)
```
Comment (chrarnoldus) [inline ValueUtils/ReflectionHelper.cs]: This method crashes with a `NullReferenceException` when `type` is an interface type. I assume this is an unsupported scenario anyway, but perhaps there should be a more specific error.

File: ValueUtilsTest/GetAllFieldTheory.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+using Xunit;
+
+namespace ValueUtilsTest
+{
+    public class GetAllFieldTheory
+    {
+        [Fact]
+        public void SampleStructHasFourFields()
+        {
+            PAssert.That(() => ReflectionHelper.GetAllFields(typeof(SampleStruct)).Select(f=>f.Name)
```
Comment (chrarnoldus) [inline ValueUtilsTest/GetAllFieldTheory.cs]: Formatting of `f=>f.Name` is not consistent in file.

File: ValueUtils/ReflectionHelper.cs (hunk 1 above)
Comment (EamonNerbonne) [inline ValueUtils/ReflectionHelper.cs]: Fixed by explicitly checking for classes.

**Pull request merged**